Setup podcast episode pipeline

rccsousa opened the pull request:
- generated graphQL schemas,
- setup fetchers,
- setup basic podcast episode page,

rccsousa committed:
Create pages for collections and graphQL schema

- scaffolding pages for new collections
- graphQL schema generation
- updating payload-types.ts for content fetching

rccsousa committed:
Setup Podcast Episode pipeline

- generated GraphQL schemas;
- created fetchers;
- created basic landing page for testing slug navigation;

rccsousa commented:
For the time being addresses #8 creating a workaround through a simplified version that fetches only published episodes and returns a 404 page if the episode is not publicly available (seems most logical).

davelange commented on src/app/_graphql/contributors.ts:
```diff
@@ -0,0 +1,21 @@
+  }
+`
+
+export const CONTRIBUTOR = `
```
Is this supposed to receive a slug to fetch a specific contributor?

rccsousa replied:
eventually, yes. i moved to podcast episodes because at the time i was struggling with auth
will get back to this once podcasts is wrapped 

(should've stuck with this until finishing it shouldn't I? :\)

rccsousa committed:
update podcast-episode pipeline and new hooks

- created contributor hook to populate contributor field on read,
- updated graphQL queries

davelange commented on src/app/_api/fetchPodcastEpisodes.ts:
```diff
@@ -0,0 +1,35 @@
+  // }
+
+  const doc = await fetch(`${GRAPHQL_API_URL}/api/graphql`, {
+    method: "POST",
```
If you're going to do this often, it might be better to create a helper function that receives the graphql query and deals with the boilerplate of setting method, headers, etc. Then you can just reuse it like `await request(PODCAST_EPISODES)`, for all the different queries

rccsousa replied:
great idea, will encapsulate this

davelange commented on src/app/_graphql/podcastepisodes.ts:
```diff
@@ -0,0 +1,55 @@
+// TODO: Figure out refering to contributors and media files.
+
```
Rename to `podcastEpisodes.ts`, to keep consistency ;)

rccsousa replied:
done ✅

davelange commented on src/payload/collections/PodcastEpisodes/hooks/populateContributors.ts:
```diff
@@ -0,0 +1,25 @@
+import type { AfterReadHook } from "payload/dist/collections/config/types";
+
+export const populateContributors: AfterReadHook = async ({ doc, req: { payload } }) => {
```
Why do we need this? It would be better to deal with this in a seperate PR, where you can explain better why it's needed/what it does

rccsousa replied:
this hook is similar to populateAuthors for the posts 

why it's there: 

- contributors are not public but this hook fetches exposes the contributor information to the field populatedContributors, it's the way to show the author/contributor without exposing the collection 
 
an alternative would be exposing contributors (that does not have sensitive info) and skip this altogether, which seems the best approach?

davelange replied:
I don't think contributors needs to be private. They are basically authors and we wont be keeping any sensitive info on them

rccsousa replied:
i agree, i'll change read access to "anyone"

rccsousa replied:
done ✅

rccsousa committed:
fix: renaming and updating access control

- renamed podcastepisodes.ts --> podcastEpisodes.ts @_graphql folder for consistency
- changed access control for Contributors collection, making it publicly accessible

rccsousa committed:
fix: update PodcastEpisodes collection
